Modules/_sqlite/connection.c

```diff
@@ -1442,6 +1457,169 @@ pysqlite_connection_iterdump(pysqlite_Connection* self, PyObject* args)
     return retval;
 }
 
+#ifdef HAVE_BACKUP_API
+static PyObject *
+pysqlite_connection_backup(pysqlite_Connection *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *target = NULL;
+    const char *filename;
+    int pages = -1;
+    PyObject *progress = Py_None;
+    char *name = "main";
+    int rc;
+    int cberr = 0;
+    double sleep_secs = 0.250;
+    sqlite3 *bckconn;
+    sqlite3_backup *bckhandle;
+    static char *keywords[] = {"target", "pages", "progress", "name", "sleep", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|$iOsd:backup", keywords,
+                                     &target, &pages, &progress, &name, &sleep_secs)) {
+        return NULL;
+    }
+
+    if (!PyUnicode_Check(target) && !PyObject_TypeCheck(target, &pysqlite_ConnectionType)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "target argument must be either a string or a Connection instance");
+        return NULL;
+    }
+
+    if (progress != Py_None && !PyCallable_Check(progress)) {
+        PyErr_SetString(PyExc_TypeError, "progress argument must be a callable");
+        return NULL;
+    }
+
+    if (pages == 0) {
+        pages = -1;
+    }
+
+    if (PyUnicode_Check(target)) {
+        filename = PyUnicode_AsUTF8(target);
+        if (filename == NULL) {
+            return NULL;
+        }
+
+        Py_BEGIN_ALLOW_THREADS
+        rc = sqlite3_open(filename, &bckconn);
+        Py_END_ALLOW_THREADS
+
+        if (rc != SQLITE_OK) {
+            if (bckconn) {
+                _pysqlite_seterror(bckconn, NULL);
+                Py_BEGIN_ALLOW_THREADS
+                SQLITE3_CLOSE(bckconn);
+                Py_END_ALLOW_THREADS
+            } else {
+                (void)PyErr_NoMemory();
+            }
+            return NULL;
+        }
+    }
+    else {
+        if ((pysqlite_Connection *)target == self) {
+            PyErr_SetString(PyExc_ValueError, "target cannot be the same connection instance");
+            return NULL;
+        }
+#if SQLITE_VERSION_NUMBER < 3008007
+        /* Since 3.8.7 this is already done, per commit
+           https://www.sqlite.org/src/info/169b5505498c0a7e */
+        if (!sqlite3_get_autocommit(((pysqlite_Connection *)target)->db)) {
+            PyErr_SetString(pysqlite_OperationalError, "target is in transaction");
+            return NULL;
+        }
+#endif
+        filename = NULL;
+        bckconn = ((pysqlite_Connection *)target)->db;
+    }
+
+    Py_BEGIN_ALLOW_THREADS
+    bckhandle = sqlite3_backup_init(bckconn, "main", self->db, name);
```

Comment:
Is it worth to make the destination table name configurable?

Reply:
The second parameter, `"main"`, is the destination _database_ name (see [here](http://www.sqlite.org/c3ref/backup_finish.html#sqlite3backupinit)): in principle yes, it could be exposed as a parameter, but while the source name (ie the fourth parameter) is useful because you may desire to take a snapshot of an _attached_ database, using a target different from `"main"` seems seldom useful...